lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -983,15 +979,18 @@ private function getLazyLoadedPublicProperties(ClassMetadata $class)
 
     /**
      * @param \ReflectionParameter[] $parameters
+     * @param array<int, string>     $renameParameters
```

Comment:
Also, it looks like this is needed for magic methods and only for those. Why is that?

Reply:
I think we should stick with `array<int, string>`, `list` is way too non-standard, even for these. Use more fancy things with `@psalm-param` only.

Reply:
isn't `string[]` kind of the accepted notation for this?

Reply:
`string[]` is also fine in my view, but OP went with more specific `array<int, string>`, which is also fine. I just don't want to bother people to change unnecessary things, it's too meaningless even for my taste. It would be more important if this was `array<int, SomeClass>`, because unlike `Someclass[]` PHPStorm does not provide autocompletion for that out of the box when iterating over it. But `array<int, string>` vs `string[]`, I think doesn't make a difference in PHPStorm's view.